src/object/cli_ec.c

```diff
@@ -1918,10 +1918,20 @@ obj_ec_fetch_set_sgl(struct dc_object *obj, struct obj_reasb_req *reasb_req,
 				tail_hole_size += recx_size;
 				continue;
 			}
+
 			D_ASSERT(data_size <= recx_size);
 			tail_hole_size += recx_size - data_size;
 			D_ASSERT(tail_hole_size <= size_in_iod);
-			dc_sgl_out_set(usgl, size_in_iod - tail_hole_size);
+
+			/**
+			 * During EC data recovery, data size might be shorter than
+			 * the real data size, because the tail degraded sgl might
+			 * be truncated by ioc_trim_tail_holes. so the sgl buf size
+			 * is set by obj_ec_recov_fill_back().
+			 */
+			if (!reasb_req->orr_recov_data ||
```

Comment:
"reasb_req->orr_recov_data" looks need not to add such a new flag, can pass the "data_recov" flag in obj_ec_comp_cb() to obj_ec_fetch_set_sgl().

Reply:
orr_recov means the shard needs to be rebuilt from parity, Orr_recov_data means the data is actually rebuilt from the parity shard. 

And also Orr_recov is set by the recover task, but this is original task, I thought they are using different obj_auxi and rehabs_req. no?

Reply:
I mean can reuse/pass in obj_ec_comp_cb()'s "bool data_recov" flag as they should be same purpose, can be refined later. will give a +1 soon and let's rethink sgl set related things later, thx.